topic-operator/src/main/java/io/strimzi/operator/topic/TopicStoreTopologyProvider.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic;
+
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.ForeachAction;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.Stores;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.function.Supplier;
+
+/**
+ * Kafka Streams topology provider for TopicStore.
+ */
+public class TopicStoreTopologyProvider implements Supplier<Topology> {
+    private final String storeTopic;
+    private final String topicStoreName;
+    private final Properties kafkaProperties;
+    private final ForeachAction<? super String, ? super Integer> dispatcher;
+
+    public TopicStoreTopologyProvider(
+            String storeTopic,
+            String topicStoreName,
+            Properties kafkaProperties,
+            ForeachAction<? super String, ? super Integer> dispatcher
+    ) {
+        this.storeTopic = storeTopic;
+        this.topicStoreName = topicStoreName;
+        this.kafkaProperties = kafkaProperties;
+        this.dispatcher = dispatcher;
+    }
+
+    @Override
+    public Topology get() {
+        StreamsBuilder builder = new StreamsBuilder();
+
+        // Simple defaults
+        Map<String, String> configuration = new HashMap<>();
+        configuration.put(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_COMPACT);
+        configuration.put(TopicConfig.MIN_COMPACTION_LAG_MS_CONFIG, "0");
+        configuration.put(TopicConfig.SEGMENT_BYTES_CONFIG, String.valueOf(64 * 1024 * 1024));
```

Comment:
genuine question ... what's the rationale behind this value?

Reply:
Nothing more then what the javadoc says.
I saw this being used in most real examples + I remember we used this setting in our previous prod code as well.
Never really thought too much since the first usage.
@tombentley any thoughts on this?

Reply:
The default is 1GiB, which would be more than we needed, since it's a compacted topic anyway. Whether 64MiB is the _best_ default, who knows?